tests/simulator/pybind.cpp

```diff
@@ -0,0 +1,110 @@
+
+#include <pybind11/functional.h>
+#include <pybind11/pybind11.h>
+
+#include "phare/include.h"
+
+namespace py = pybind11;
+
+namespace PHARE
+{
+class StreamAppender : public SAMRAI::tbox::Logger::Appender
+{
+public:
+    StreamAppender(std::ostream* stream) { d_stream = stream; }
+    void logMessage(const std::string& message, const std::string& filename, const int line)
+    {
+        (*d_stream) << "At :" << filename << " line :" << line << " message: " << message
+                    << std::endl;
+    }
+
+private:
+    std::ostream* d_stream;
+};
+
+class SamraiLifeCycle
+{
+public:
+    static SamraiLifeCycle& INSTANCE()
+    {
+        static SamraiLifeCycle i;
+        return i;
+    }
+
+    SamraiLifeCycle()
+    {
+        SAMRAI::tbox::SAMRAI_MPI::init(0, nullptr);
+        SAMRAI::tbox::SAMRAIManager::initialize();
+        SAMRAI::tbox::SAMRAIManager::startup();
+
+        std::shared_ptr<SAMRAI::tbox::Logger::Appender> appender
+            = std::make_shared<StreamAppender>(StreamAppender{&std::cout});
+        SAMRAI::tbox::Logger::getInstance()->setWarningAppender(appender);
+    }
+    ~SamraiLifeCycle()
+    {
+        SAMRAI::tbox::SAMRAIManager::shutdown();
+        SAMRAI::tbox::SAMRAIManager::finalize();
+        SAMRAI::tbox::SAMRAI_MPI::finalize();
+    }
+
+    void reset()
+    {
+        PHARE::initializer::PHAREDictHandler::INSTANCE().stop();
+        if (sim_)
+            delete sim_;
+        sim_ = nullptr;
+        if (rdi_)
+            rdi_.reset();
+        SAMRAI::tbox::SAMRAIManager::shutdown();
+        SAMRAI::tbox::SAMRAIManager::startup();
+    }
+
+    std::unique_ptr<ISimulator, py::nodelete> getISimulator()
+    {
+        auto simulation = PHARE::getSimulator();
+        sim_            = simulation.get();
+        std::unique_ptr<ISimulator, py::nodelete> ptr{sim_};
```

Comment:
could you explain the use of py::nodelete and why do we keep the address of the simulator as a raw pointer instead of having, say, a shared_ptr

Reply:
this was the only thing I found that worked

Reply:
because? if you do something that is not "straightforward" to understand, and you had to look for a solution, please document the issue, explain (and link to) the solution?

Reply:
because the other things didn't work?
I've empirically tested this. There's a chances that other things could work, but this does.
There is no solution to link to. I've found this solution.

Reply:
feel free to test the alternatives

Reply:
assessed